src/test/java/io/camunda/zeebe/bpmnassert/testengine/db/InMemoryZeebeDbTransactionTest.java

```diff
@@ -0,0 +1,479 @@
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.bpmnassert.testengine.db;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.db.*;
+import io.camunda.zeebe.db.impl.DbLong;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public final class InMemoryZeebeDbTransactionTest {
+
+  private final ZeebeDbFactory<ColumnFamilies> dbFactory = new InMemoryZeebeDbFactory();
+
+  private TransactionContext transactionContext;
+
+  private ColumnFamily<DbLong, DbLong> oneColumnFamily;
+  private ColumnFamily<DbLong, DbLong> twoColumnFamily;
+  private ColumnFamily<DbLong, DbLong> threeColumnFamily;
+
+  private DbLong oneKey;
+  private DbLong oneValue;
+  private DbLong twoValue;
+  private DbLong twoKey;
+  private DbLong threeKey;
+  private DbLong threeValue;
+
+  @BeforeEach
+  public void setup() throws Exception {
+    final ZeebeDb<ColumnFamilies> zeebeDb = dbFactory.createDb(null);
+    transactionContext = zeebeDb.createContext();
+
+    oneKey = new DbLong();
+    oneValue = new DbLong();
+    oneColumnFamily =
+        zeebeDb.createColumnFamily(ColumnFamilies.ONE, transactionContext, oneKey, oneValue);
+
+    twoKey = new DbLong();
+    twoValue = new DbLong();
+    twoColumnFamily =
+        zeebeDb.createColumnFamily(ColumnFamilies.TWO, transactionContext, twoKey, twoValue);
+
+    threeKey = new DbLong();
+    threeValue = new DbLong();
+    threeColumnFamily =
+        zeebeDb.createColumnFamily(ColumnFamilies.THREE, transactionContext, threeKey, threeValue);
+  }
+
+  @Test
+  public void shouldUseTransaction() {
+    // given
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    twoKey.wrapLong(52000);
+    twoValue.wrapLong(192313);
+
+    threeKey.wrapLong(Short.MAX_VALUE);
+    threeValue.wrapLong(Integer.MAX_VALUE);
+
+    // when
+    transactionContext.runInTransaction(
+        () -> {
+          oneColumnFamily.put(oneKey, oneValue);
+          twoColumnFamily.put(twoKey, twoValue);
+          threeColumnFamily.put(threeKey, threeValue);
+        });
+
+    // then
+    assertThat(oneColumnFamily.exists(oneKey)).isTrue();
+    assertThat(twoColumnFamily.exists(twoKey)).isTrue();
+    assertThat(threeColumnFamily.exists(threeKey)).isTrue();
+  }
+
+  @Test
+  public void shouldNotGetPreviousValue() {
+    // given
+    oneKey.wrapLong(123);
+    oneValue.wrapLong(456);
+
+    transactionContext.runInTransaction(
+        () -> {
+          oneColumnFamily.put(oneKey, oneValue);
+          oneColumnFamily.get(oneKey);
+          oneKey.wrapLong(-1);
+
+          // when
+          final DbLong zbLong = oneColumnFamily.get(oneKey);
+
+          // then
+          assertThat(zbLong).isNull();
+        });
+  }
+
+  @Test
+  public void shouldStartNewTransaction() throws Exception {
```

Comment:
🔧 This test doesn't fit the name. It puts 3 values in the database, commits them and check that they exist. It never checks if the transaction was completed and a new one has been started. The transaction might still be active, because if the keys are available in the transaction cache `exists()` will also return true.